.travis.yml

```diff
@@ -14,16 +14,26 @@ dist: trusty
 compiler: clang # per: https://docs.travis-ci.com/user/languages/c
 # https://docs.travis-ci.com/user/environment-variables
 env:
- - CXX=clang++ DEBUG='nightmare*'
- # - CXX=clang++ DISPLAY=':99.0' DEBUG='nightmare*'
-# according to travis docs, the services -xvfb line will set DISPLAY
 branches:
   only:
     - master # https://github.com/okTurtles/group-income-simple/issues/58
 addons:
   apt:
     packages:
       - xvfb
+cache:
+  # Caches $HOME/.npm when npm ci is default script command
+  directories:
+    # cache folder with Cypress binary to avoid downloading Cypress again and again
+    - ~/.cache
+install:
+  - npm ci
+script:
+  ## run backend tests
+  - npm run test
+  ## run cypress (frontend tests)
+  - npm run test:cy
```

Comment:
Note: By default, [cypress runs on Electron instead of Chrome during CI](https://docs.cypress.io/guides/guides/launching-browsers.html#Electron-Browser). Is that a concern for you?

Reply:
No, not an issue. Though this should be a single command as before, `grunt test`, instead of `npm run test:cy` and `npm run test`

Reply:
addressed in another discussion